### DISCUSSION TITLE
excution order of 'set' and 'connect'

### DISCUSSION THREAD

**Discussion by kin-for-test** [prisma/prisma#19491] [Q&A] [answered]
```
  public static async update(user) {
    let result = await prisma.user.update({
      where: { id: user.id },
      data: {
        roles: { set: [], connect: user.roles.map((e) => ({ id: e.id })) },
      },
    });
  }
```

what is the result of this code?
set first  or connect first?

if the code is 
 `     
        roles: {connect: user.roles.map((e) => ({ id: e.id })), set: [] },
`
will it have a different result?

**Comment by nurul3101** [comment]
Hello @kin-for-test 👋 
Thanks for asking such an insightful question 😄 

I just tested it, in both cases the output would be different.

In first case (set before connect), you would see that user is connected with all the roles from the roles array.
While in the second case(set after connect), you would see that user is not connected with any of the roles as `set:[]` would clear all the connected records.

**Reply by kin-for-test** [reply]
is that exist a priority list in prisma doc to clearly defined the order of  operator?

**Reply by nurul3101** [reply, marked as answer]
There isn't a list at the moment, I'll discuss this with the team so that we can better document this.
Thanks for raising this 🙌
